metrics/prometheus/collector.go

```diff
@@ -98,6 +102,19 @@ func (c *collector) addResettingTimer(name string, m metrics.ResettingTimer) {
 	c.buff.WriteRune('\n')
 }
 
+func (c *collector) writeGaugeInfo(name string, value metrics.GaugeInfoValue) {
+	name = mutateKey(name)
+	c.buff.WriteString(fmt.Sprintf(typeGaugeTpl, name))
+	c.buff.WriteString(fmt.Sprintf("%s {", name))
+	for idx, entry := range value {
+		c.buff.WriteString(fmt.Sprintf("%s=\"%s\"", entry.Key, entry.Val))
+		if idx != len(value)-1 {
+			c.buff.WriteString(", ")
+		}
+	}
+	c.buff.WriteString("} 1 \n\n")
```

Comment:
Fragile again:
```
c.buff.WriteString(fmt.Sprintf("%s {", name))
entries := make([]string, 0, len(value))
for _, entry := range value {
        entries = append(entries, fmt.Sprintf("%s=%q", entry.Key, entry.Val))
}
c.buff.WriteString(fmt.Sprintf("{%s}",strings.Join(entries,",")))
c.buff.WriteString("} 1 \n\n")
```

Reply:
Let's just avoid writing json manually, we use the `json` package for that